x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/cache/blob/SearchableSnapshotsBlobStoreCacheMaintenanceIntegTests.java

```diff
@@ -236,6 +201,134 @@ public void testMaintenance() throws Exception {
         });
     }
 
+    /**
+     * Test that obsolete blob cache entries are deleted from the system index by the periodic maintenance task.
+     */
+    public void testPeriodicMaintenance() throws Exception {
+        ensureStableCluster(internalCluster().getNodeNames().length, TimeValue.timeValueSeconds(60L));
+
+        createRepository("repo", FsRepository.TYPE);
+        Map<String, Tuple<Settings, Long>> mountedIndices = mountRandomIndicesWithCache("repo", 1, 3);
+        ensureYellow(SNAPSHOT_BLOB_CACHE_INDEX);
+
+        final long nbEntriesInCacheForMountedIndices = mountedIndices.values().stream().mapToLong(Tuple::v2).sum();
+        refreshSystemIndex(true);
+        assertThat(numberOfEntriesInCache(), equalTo(nbEntriesInCacheForMountedIndices));
+
+        createRepository("other", FsRepository.TYPE);
+        Map<String, Tuple<Settings, Long>> otherMountedIndices = mountRandomIndicesWithCache("other", 1, 3);
+
+        final long nbEntriesInCacheForOtherIndices = otherMountedIndices.values().stream().mapToLong(Tuple::v2).sum();
+        refreshSystemIndex(true);
+        assertThat(numberOfEntriesInCache(), equalTo(nbEntriesInCacheForMountedIndices + nbEntriesInCacheForOtherIndices));
+
+        final int oldDocsInCache;
+        if (randomBoolean()) {
+            oldDocsInCache = indexRandomDocsInCache(1, 50, Instant.now().minus(Duration.ofDays(7L)).toEpochMilli());
+            refreshSystemIndex(true);
+            assertThat(
+                numberOfEntriesInCache(),
+                equalTo(nbEntriesInCacheForMountedIndices + nbEntriesInCacheForOtherIndices + oldDocsInCache)
+            );
+        } else {
+            oldDocsInCache = 0;
+        }
+
+        // creates a backup of the system index cache to be restored later
+        createRepository("backup", FsRepository.TYPE);
+        createSnapshot("backup", "backup", List.of(SNAPSHOT_BLOB_CACHE_INDEX));
+
+        final Set<String> indicesToDelete = new HashSet<>(mountedIndices.keySet());
+        indicesToDelete.add(randomFrom(otherMountedIndices.keySet()));
+
+        assertAcked(systemClient().admin().indices().prepareDelete(SNAPSHOT_BLOB_CACHE_INDEX));
+        assertAcked(client().admin().indices().prepareDelete(indicesToDelete.toArray(String[]::new)));
+        assertAcked(client().admin().cluster().prepareDeleteRepository("repo"));
+        ensureClusterStateConsistency();
+
+        refreshSystemIndex(false);
```

Comment:
Is this line necessary?

Reply:
Looks like a left over from debugging the test. It is not required as the system index is deleted before. I removed it in  f963685226a46351563dd8efc6b56b86021dccc4